### PR TITLE
Adding StopArea extraction and route_master filter

### PR DESCRIPTION
**Pull request by prhod**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -156,34 +166,60 @@ fn is_stop_point(obj: &osmpbfreader::OsmObj) -> bool {
     obj.node().and_then(|n| n.tags.get("highway")).map_or(false, |v| v == "bus_stop")
 }
 
+fn is_stop_area(obj: &osmpbfreader::OsmObj) -> bool {
+    obj.tags().get("public_transport").map_or(false, |v| v == "stop_area") &&
+    obj.is_relation()
```
[the remark this answers is not in the document]
Comment (prhod) [reply src/lib.rs]: thanks for the hint, I made the change !

File: src/lib.rs [hunk 2]
```diff
@@ -156,34 +166,60 @@ fn is_stop_point(obj: &osmpbfreader::OsmObj) -> bool {
...
+                                  "power",
+                                  "proposed",
+                                  "road",
+                                  "running",
+                                  "ski",
+                                  "historic"];
+    obj.relation().and_then(|r| r.tags.get("type")).map_or(false, |v| v == "route_master") &&
+    obj.relation()
+        .and_then(|r| r.tags.get("route_master"))
+        .map_or(false, |v| !non_pt_route_type.contains(&v.as_str()))
```
[the remark this answers is not in the document]
Comment (prhod) [reply src/lib.rs]: I'm checking with @nlehuby, there could be a need for non regular values to be extracted. TO ease de PR, I made a white list but he black list is kept in commentary.

File: src/lib.rs [hunk 3]
```diff
@@ -156,34 +166,60 @@ fn is_stop_point(obj: &osmpbfreader::OsmObj) -> bool {
...
+                                  "mtb",
+                                  "nordic_walking",
+                                  "pipeline",
+                                  "piste",
+                                  "power",
+                                  "proposed",
+                                  "road",
+                                  "running",
+                                  "ski",
+                                  "historic"];
```
[the remark this answers is not in the document]
Comment (prhod) [reply src/lib.rs]: I could'nt create a global variable, so I made a function. Please tell me if there is a cleaner way of doing it ;)

[the remark this answers is not in the document]
Comment (TeXitoi) [reply src/lib.rs]: That's better with a function. I like it like that.

**Comment by prhod** [comment]
sorry, seems I forgot to push the last modification :'( I'll do it later today !

**Comment by prhod** [comment]
I previously made the push on my fork instead of this branch :p sorry for that !

**Comment by prhod** [comment]
thanks !

**Pull request merged**